### PR TITLE
Show created targets

### PR DESCRIPTION
**Pull request by agustito37**
#### Description
As a user I should be able to see all my targets in the map so I can easily identify them

#### Tasks
- The map should be centred on the Intendencia de Montevideo by default
- The map should show all "el centro"
- Given that there are existing targets created in the app, I should be able to see them marked on the map
- I should see a different icon for each topic
- I should see a different radius size depending on the target size

### CODE REVIEW

File: app/controllers/targets_controller.rb [hunk 1]
```diff
@@ -25,4 +24,9 @@ def create
   def target_params
     params.require(:target).permit(:area, :title, :topic_id, :latitude, :longitude)
   end
+
+  def common_data
```
Comment (matiasmansilla1989) [inline app/controllers/targets_controller.rb]: Controller filters should be used to halt the request cycle and not to load data. 
You can use a simple method instead or a `helper_method` if you want to call them from the view.
Check:
- http://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html
- https://apidock.com/rails/ActionController/Helpers/ClassMethods/helper_method

File: app/assets/javascripts/base/_g_map.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
+var GMap = (() => {
+	var MEDIATOR_NAME = "g_map";
```
Comment (fedeagripa) [inline app/assets/javascripts/base/_g_map.js]: Indentation

File: app/assets/javascripts/base/_g_map.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  var MARKERS_HEIGHT = 20;
+  var MARKERS_COLOR = '#edcc00';
+  var MARKERS_AREA_FACTOR = 1;
+  var ANCHOR_X = 28.5;
+  var ANCHOR_Y = 50;
+  var MAP_SELECTOR = '#map';
+  var _map;
+  var _marker;
+
+	Mediator.installTo(this);
```
Comment (fedeagripa) [inline app/assets/javascripts/base/_g_map.js]: same here

File: app/assets/javascripts/base/_g_map.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      radius: radius * MARKERS_AREA_FACTOR,
+      fillColor: MARKERS_COLOR,
+      strokeWeight: 0
+    });
+
+    circle.bindTo('center', marker, 'position');
+  }
+
+  function markerChanged(event) {
+    publish(MEDIATOR_NAME+".marker_changed", { lat:event.latLng.lat(), lng:event.latLng.lng() });
```
Comment (fedeagripa) [inline app/assets/javascripts/base/_g_map.js]: interpolate the constant so you don't need to concat

File: app/assets/javascripts/base/_main.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+var Main = (() => {
+  var MEDIATOR_NAME = "main";
+
+  Mediator.installTo(this);
+
+  function init() {
+    publish(MEDIATOR_NAME+".init");
```
Comment (fedeagripa) [inline app/assets/javascripts/base/_main.js]: Same here about interpolation

File: app/assets/javascripts/base/_main.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  }
+
+  function checkControllerAction(controller, action) {
+    return $("body[data-controller='" + controller + "'][data-action='" + action + "']").length > 0
```
Comment (fedeagripa) [inline app/assets/javascripts/base/_main.js]: Interpolate so you don't need to switch between single & double quotes

File: app/assets/javascripts/views/_new_form.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+var NewForm = (() => {
```
Comment (fedeagripa) [inline app/assets/javascripts/views/_new_form.js]: check single or double quotes consinstency and interpolation in the file

File: app/assets/stylesheets/base/_main.scss [hunk 8]
```diff
@@ -25,7 +25,7 @@ html,body {
   }
 
   h2 {
-    font-size: 12px;
+    font-size: .75rem;
```
Comment (fedeagripa) [inline app/assets/stylesheets/base/_main.scss]: If you are switching to rem please do it in all the possible cases (maybe do it in another pr if you wish to)

File: app/assets/javascripts/base/_g_map.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+	Mediator.installTo(this);
+
+  // Init after Main init
+  subscribe("main.init",init);
+
+  function init() {
+    createMap();
+    bindEvents();
+    publish(MEDIATOR_NAME+".init",{ lat:DEFAULT_LAT, lng:DEFAULT_LNG });
```
Comment (grilix) [inline app/assets/javascripts/base/_g_map.js]: Be careful with the spaces, for example around `+` operator, after `,`, etc. There are a couple of examples where spacing is not consistent.

File: app/assets/javascripts/base/_g_map.js [hunk 10]
```diff
@@ -0,0 +1,121 @@
+var GMap = (() => {
+	var MEDIATOR_NAME = "g_map";
+  var DEFAULT_LAT = -34.906649769289196;
+  var DEFAULT_LNG = -56.18607223033905;
+  var DEFAULT_ZOOM = 16;
+  var MARKER_WIDTH = 57;
+  var MARKER_HEIGHT = 72;
+  var MARKERS_WIDTH = 20;
+  var MARKERS_HEIGHT = 20;
+  var MARKERS_COLOR = '#edcc00';
```
Comment (grilix) [inline app/assets/javascripts/base/_g_map.js]: Always use double quotes for strings (unless the standard is to always use single quotes, but don't mix them.)

**Comment by agustito37** [comment]
Removed the before_action sentence in targets controller, added string interpolation in js files so the "+" is not used anymore and fixed indentation problems.
